### PR TITLE
Alter hook_task TEXT fields to LONGTEXT

### LINKED ISSUE #10252: When commit too many files, then return errors
```
remote: Gitea: Internal Server Error
remote: Failed to Update: bigdata/cleaner Branch: lunny/go_mod Error: CommitRepoAction: PrepareWebhooks: CreateHookTask: Error 1406: Data too long for column 'payload_content' at row 1
```

### PR DESCRIPTION
**Pull request by 42wim**
Mysql TEXT has a limit of 64KB, change this to LONGTEXT in mysql only so we can have bigger hook payloads.

Postgresql has unlimited TEXT - https://www.postgresql.org/docs/current/datatype-character.html
Sqlite has unlimited TEXT - https://www.sqlitetutorial.net/sqlite-data-types/#:~:text=The%20maximum%20length%20of%20TEXT,SQLite%20supports%20various%20character%20encodings.

Same issue as #16656 but for hook_task

Fixes #10252, #19679, #3561

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
I'd probably argue this should be backported to 1.17 too.

File: models/migrations/v217.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+func alterHookTaskTextFieldsToLongText(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if setting.Database.UseMySQL {
```
Comment (lunny) [inline models/migrations/v217.go:20]: Why MySQL only?

Comment (zeripath) [reply models/migrations/v217.go:20]: v191 only runs for mysql too.

Comment (42wim) [reply models/migrations/v217.go:20]: I've updated the PR message to contain the necessary info about that.

**Review by 6543 — COMMENTED** [review]
lgtm

**Pull request merged**